scrapy/spidermiddlewares/start.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from scrapy.http import Request
+
+if TYPE_CHECKING:
+    from collections.abc import AsyncIterator, Iterable
+
+    from scrapy import Spider
+
+
+class StartSpiderMiddleware:
+    """Set ``"is_start_request"`` to ``True`` in the
+    :attr:`~scrapy.Request.meta` of :ref:`start requests <start-requests>`.
```

Comment:
I think we should document the new meta key in the release notes, when mentioning that this new middleware is enabled by default.

We should also document it in https://docs.scrapy.org/en/latest/topics/request-response.html#request-meta-special-keys.